.github/workflows/codeql.yml

```diff
@@ -0,0 +1,63 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ 'master' ]
+  pull_request:
+    branches: [ 'master' ]
+  schedule:
+    - cron: '39 18 * * 0'
+
+env:
+  UBUNTU_DEP_BUILD: |
+    gcc g++ pkg-config systemd libevdev-dev libglib2.0-dev
+    libjson-glib-dev libsystemd-dev libudev-dev libunistring-dev
+    python3-dev python3-evdev swig python3-sphinx
+  UBUNTU_DEP_TEST: check python3-gi python3-lxml valgrind
+  PIP_PACKAGES: meson ninja libevdev pyudev pytest sphinx_rtd_theme
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-20.04
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp', 'python' ]
+        meson_args:
+          - ''
+          - '-Dbuildtype=plain'
+          - '-Dbuildtype=release'
+          - '-Ddocumentation=true'
```

Comment:
Does it really need to run analyzing on every build type, especially for Python, where I don't think it has any difference?